content/knowledge/consensus_algorithms.md

```diff
@@ -0,0 +1,367 @@
+In a blockchain network nodes repeatedly decide what the complete history of a shared ledger is, from among multiple possible versions that occasionally conflict between each other.
+This network-wide agreement allows the recipient of a crypto coin to have faith that the coin is both valid (not counterfeit) and not already spent elsewhere.
+
+### Federated Byzantine Agreement
+
+#### Introduction
+
+Any agreement system in a distributed computing network needs to be fault-tolerant.
+This means it must produce consistent results despite errors like slow communication links, unresponsive nodes and misordered messages.
+A Byzantine agreement system is additionally tolerant of [`Byzantine faults`](https://en.wikipedia.org/wiki/Byzantine_fault): nodes that give false information, whether due to error or in a deliberate attempt to subvert the system or gain some advantage.
+Such a system prevents the double spending problem by using a form of majority called _quorum_.
+A node in such a network refuses to commit to a particular version of the ledger history until it sees that enough of its peers are also prepared to commit to it.
+The number of peers that is enough to make a decision is the quorum and once each of the peers commit to a value after its quorum had committed, the system form a large enough voting bloc to force the remaining nodes to agree with the decision.
+In such a system, the attempts from malicious nodes to create double spending events, or in other way invalid transactions, are overwhelmed by the network votes from honest nodes.
+
+For a majority to be formed, the number of all nodes in the system must be known and its quorum calculated at any time.
+Opposed to this concept, in a Federated Byzantine Agreement System (FBAS) the participants in the network are loosely defined and can join and leave at any time without coordinating with any central authority.
+In such a system, quorums are incomplete membership snapshots that can change at any time.
+This makes the system consensus probabilistic with some error size (probability of not reaching a consensus) present at all time.
+
+The network does not trust a single entity to operate the network, but rather works as a decentralized system, in which each participant knows only about a group of others peers, but not about all the nodes in the network.
+In the FBAS, each node has knowledge of other nodes that it considers important.
+When making a decision, it waits for these important nodes to agree on a value before considering a transaction settled.
+
+Each node of that group also has knowledge about its own list of important nodes and considers their statement on a value before deciding on a transaction.
+Also each node is potentially a member of a group of nodes, that another considers a benchmarking group.
+On a higher level, the structure looks like multiple overlapping groups of participants.
```

Comment:
We could add that this is where the term `federated` in the name comes from, because it's a federation of overlapping groups.